### PR TITLE
Add conditional check to replace launch-site task with LYS task

### LINKED ISSUE #1497: [LYS] Replace launch-site task with LYS
### Issue description

P2: https://wp.me/pfwcF5-Cs

Replace the existing [launch-site task](https://github.com/Automattic/wc-calypso-bridge/blob/d2da96a29f3ade92228b10484897f79539d1f659/includes/tasks/class-wc-calypso-task-launch-site.php#L13) with WooCommerce's LYS.

Image

 
### Acceptance criteria

1. In all plans except free trial, WooCommerce LYS task should be shown.
2. Add a redirect from `/wp-admin/admin.php?page=wc-admin&task=launch_site` to LYS task
3. LYS task in WPCOM should work exactly the same as other WooCommerce sites.

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes #1497

- Replace `launch-site` task with LYS task when LYS feature is enabled. 
- Refactor and merge `add_tasks` function to `replace_tasks` to consolidate logic and enable easier tasks sorting

I decided to skip adding a redirect from `/wp-admin/admin.php?page=wc-admin&task=launch_site` because it doesn't seem to be a simple endeavour and it only serves a tiny edge case like users having the link opened. 

This PR does not handle free trial detection yet.

#### Test LYS feature enabled
1. Use a WPCOM atomic e-commerce site
1. Enable LYS feature flag (via beta tester or you can set [these lines](https://github.com/Automattic/wc-calypso-bridge/blob/507bde7b933c60bb2cb7efdfad66b7bd8581b39b/includes/class-wc-calypso-bridge-woocommerce-admin-features.php#L143-L147) to `true`).
1. Go to Homescreen
2. Observe LYS task is displayed and `launch-site` task is hidden
3. Observe LYS task is displayed after `Add a domain` task

#### Test LYS feature disabled

1. Disable LYS feature flag (via beta tester or you can set [these lines](https://github.com/Automattic/wc-calypso-bridge/blob/507bde7b933c60bb2cb7efdfad66b7bd8581b39b/includes/class-wc-calypso-bridge-woocommerce-admin-features.php#L143-L147) to `false`).
1. Go to Homescreen
2. Observe `launch-site` task is displayed and LYS task is hidden
3. Observe `launch-site` task is displayed after `Add a domain` task

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: includes/class-wc-calypso-bridge-setup-tasks.php [hunk 1]
```diff
@@ -86,14 +85,9 @@ public function handle_ajax_launch_endpoint() {
 	}
 
 	/**
-	 * Add Setup Tasks.
+	 * Add and replace setup tasks.
 	 */
-	public function add_tasks() {
```
Comment (ilyasfoo) [inline includes/class-wc-calypso-bridge-setup-tasks.php:91]: Entirely merged this function into `replace_tasks`

File: includes/class-wc-calypso-bridge-setup-tasks.php [hunk 2]
```diff
@@ -141,6 +115,21 @@ public function replace_tasks( $lists ) {
...
+							// Append add domain task before launch your store task.
+							require_once __DIR__ . '/tasks/class-wc-calypso-task-add-domain.php';
+							$add_domain_task = array( new \Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\AddDomain( $lists['setup'] ) );
+							array_splice( $lists['setup']->tasks, $index, 0, $add_domain_task );
+
+							// Replace launch your store task with launch site task.
+							if ( ! Features::is_enabled( 'launch-your-store' ) ) {
+								require_once WC_CALYPSO_BRIDGE_PLUGIN_PATH . '/includes/tasks/class-wc-calypso-task-launch-site.php';
+								$launch_site_task = new \Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\LaunchSite( $lists['setup'] );
+								$lists['setup']->tasks[$index + 1] = $launch_site_task;
```
Comment (ilyasfoo) [inline includes/class-wc-calypso-bridge-setup-tasks.php:129]: `$index + 1` because index was bumped up since we added domain task above

**Comment by psealock** [comment]
>Test LYS feature disabled
Disable LYS feature flag (via beta tester or you can set [these lines](https://github.com/Automattic/wc-calypso-bridge/blob/507bde7b933c60bb2cb7efdfad66b7bd8581b39b/includes/class-wc-calypso-bridge-woocommerce-admin-features.php#L143-L147) to true).

@ilyasfoo Can you explain this further? Do you mean leave them uncommented or set `$features['launch-your-store'] = true;`?

If the feature flag is set to true, then I get the same results as the first section, which makes sense.

image

If I leave that code as-is, I get the following:

image

**Comment by ilyasfoo** [comment]
> @ilyasfoo Can you explain this further? Do you mean leave them uncommented or set $features['launch-your-store'] = true;?
@psealock Sorry that was a copy-pasta error. It should be leave the code as-is, fixed it in description

**Review by chihsuan — APPROVED** [review]
LGTM and tested well! 👍

**Comment by psealock** [comment]
>Observe launch-site task is displayed after Add a domain task

I'm seeing that I've already launched

image

So I follow the link and go back coming soon mode, but the task list is the same. Should I be seeing something else? I'm wondering if there is a sync issue.

image

**Comment by ilyasfoo** [comment]
> So I follow the link and go back coming soon mode, but the task list is the same. Should I be seeing something else? I'm wondering if there is a sync issue.

@psealock Good question! This is expected.

The `launch-site` task checks for [launch status](https://github.com/Automattic/wc-calypso-bridge/blob/27842c743b414e047e46423f9cd132fca58eff16/includes/tasks/class-wc-calypso-task-launch-site.php#L86), not coming soon status. WPCOM has a separate launch status (`launch-status` option) and coming soon status (`blog_public`, `wpcom_public_coming_soon` and `wpcom_coming_soon` options)

**Review by psealock — APPROVED** [review]
Thanks for the  clarification @ilyasfoo, this is all working well then. 🚢
